src/base/object_representation.h

```diff
@@ -694,20 +684,13 @@
   ((OR_GET_INT(((char *) (ptr)) + OR_REP_OFFSET)) \
    & OR_MVCC_REPID_MASK)
 
-/* in MVCC, chn follow by rep_id and/or ins_id depending by flags */
-#define OR_GET_MVCC_CHN(ptr, mvcc_flags) \
-  (((mvcc_flags) & OR_MVCC_FLAG_VALID_DELID) \
-   ? NULL_CHN \
-   : (((mvcc_flags) & OR_MVCC_FLAG_VALID_INSID) \
-      ? (OR_GET_INT (((char *) (ptr)) + OR_REP_OFFSET + OR_MVCC_REP_SIZE + OR_MVCCID_SIZE)) \
-      : ((OR_GET_INT (((char *) (ptr)) + OR_REP_OFFSET + OR_MVCC_REP_SIZE)))))
-
+/* in MVCC, chn follow by rep_id, ins_id or del_id depending by flags */
 #define OR_GET_MVCC_CHN_OFFSET(mvcc_flags) \
-  (((mvcc_flags) & OR_MVCC_FLAG_VALID_DELID) \
-   ? -1 \
-   : (((mvcc_flags) & OR_MVCC_FLAG_VALID_INSID) \
-      ? (OR_REP_OFFSET + OR_MVCC_REP_SIZE + OR_MVCCID_SIZE) \
-      : (OR_REP_OFFSET + OR_MVCC_REP_SIZE)))
+  (OR_REP_OFFSET + OR_MVCC_REP_SIZE + ((mvcc_flags) & OR_MVCC_FLAG_VALID_INSID ? OR_MVCCID_SIZE : 0) \
+  + (((mvcc_flags) & OR_MVCC_FLAG_VALID_INSID) ? OR_MVCCID_SIZE : 0))
```

Comment:
@funkydvd 

I think the second `OR_MVCC_FLAG_VALID_INSID` should be `OR_MVCC_FLAG_VALID_DELID`.